spec/development/workload/PageIterator.ts

```diff
@@ -0,0 +1,61 @@
+import { assert } from "chai";
+import { Event } from "microsoft-graph";
+
+import { PageIterator, PageIteratorCallback } from "../../../src/tasks/PageIterator";
+import { getClient } from "../test-helper";
+
+const client = getClient();
+describe("PageIterator", function() {
+	describe("sameHeadersPassedWithPageIterator", () => {
+		const pstHeader = { Prefer: 'outlook.timezone= "pacific standard time"' };
+		const utc = "UTC";
+		const pst = "Pacific Standard Time";
+
+		it("verify same headers", async () => {
+			const response = await client
+				.api(`/me/events?$top=2`)
+				.headers(pstHeader)
+				.select("id,start,end")
+				.get();
+
+			const callback: PageIteratorCallback = (eventResponse) => {
+				const event = eventResponse as Event;
+				assert.equal(event.start.timeZone, pst);
+				return true;
+			};
+			var requestOptions = { headers: pstHeader };
+			if (response["@odata.nextLink"]) {
```

Comment:
if there isn't a next link then we'll just assume things have passed but haven't actually tested the change.

Reply:
Added a before hook to make sure that certain number of events are present/created.